lib/shared/movie.dart

```diff
@@ -8,51 +11,86 @@ class Movie {
     required this.genres,
     required this.overview,
     this.saved = false,
-    this.cast = const <String>[],
+    required this.cast,
   });
 
+  final int id;
   final String title;
   final String posterName;
   final String backdropName;
   final String releaseDate;
-  final String genres;
+  final List<int> genres;
   final String overview;
   final double score;
-  final List<String> cast;
+  final List<dynamic> cast;
   bool saved;
+  static const uri = 'https://image.tmdb.org/t/p/w500';
 
   void toggleSaved() {
     saved = !saved;
   }
 
-  String get assetsBackdropPath => 'assets/$backdropName.jpg';
+  String get assetsBackdropPath => uri + backdropName;
 
-  String get assetsPosterPath => 'assets/$posterName.jpg';
+  String get assetsPosterPath => uri + posterName;
 
   List<String> get assetsCastPath {
     List<String> castPath = [];
     for (int i = 0; i < cast.length; i++) {
-      castPath.add('assets/cast/${cast[i]}.jpg');
+      castPath.add(uri + cast[i]["profile_path"]!);
     }
     return castPath;
   }
 
+  Future<String> get getGenres => genresToString(genres);
+
+  Future<String> genresToString(List<dynamic> list) async {
+    MyLocalRepository provider = MyLocalRepository();
+    final List<String> genreNames = await provider.getGenresById(genres);
+
+    String genresString = genreNames.join(", ");
+    return genresString;
```

Comment:
You shouldn't have a repository call inside a data class. Also, this method is used to show the genres in the *UI*, so it should be placed inside a widget.

Reply:
Fixed